### ISSUE TITLE
[copilot chat] Cannot copy to clipboard when code/text is still streaming

### ISSUE THREAD

**Issue by theskcd** [microsoft/vscode-copilot-release#1058]
- VS Code Version: 1.87.2
- OS Version: Darwin arm64 23.0.0
- Logs:

Steps to Reproduce:

1. Ask copilot to add comments to the open file
2. Try to select some code and copy it, it fails and the selection gets reset many times.

https://github.com/microsoft/vscode-copilot-release/assets/10356126/aa7f55bf-5af8-40e6-809e-f312141dc002

I have attached a screen-grab of what it looks like.

**Comment by theskcd** [comment]
Here are some debugging traces which might be helpful, I think whats happening is that the list-renderer is replacing the DOM node without diffing it properly at this code location: https://github.com/microsoft/vscode/blob/dd42b9fda56ee3c2f37cf94941a1220c213aca4f/src/vs/workbench/contrib/chat/browser/chatListRenderer.ts#L642

So I tried using DomDiff to apply just the diff to the element, but turns out the ID of the whole Dom node is changing causing a re-render which is triggered from here: https://github.com/microsoft/vscode/blob/dd42b9fda56ee3c2f37cf94941a1220c213aca4f/src/vs/workbench/contrib/chat/browser/chatWidget.ts#L317-L332 here the ID changes because on every update to the stream we increment the dataid here: https://github.com/microsoft/vscode/blob/dd42b9fda56ee3c2f37cf94941a1220c213aca4f/src/vs/workbench/contrib/chat/common/chatViewModel.ts#L353-L355

I am not sure how to solve this proper, but I think for the code blocks being generated (since they are just monaco editors in disguise) we can avoid the re-rendering in a smart way.

I stopped debugging at this point, because I did not want to spend more time understanding the re-rendering logic. Hopefully this helps!
